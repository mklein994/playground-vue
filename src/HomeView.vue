<template>
  <div class="welcome-container">
    <h1 class="welcome">Welcome</h1>
  </div>
</template>

<style scoped>
.welcome-container {
  display: grid;
  height: 100vh; /* assumes <body> has no margin */
  box-sizing: border-box; /* For when no resets are used */
  padding-bottom: 3.5rem;
  place-items: center;
}

.welcome {
  font:
    700 var(--font-size) "Andada Pro",
    serif;

  margin-block: 0;
  text-align: center;

  /* Don't let this go below a comfortable font size. */
  --min-font-size: 1rem;

  /* Based on the distance away from the edges on narrow screens. */
  --preferred-font-size: 15vw;

  /* Based on its proportions on super-wide screens. */
  --max-font-size: 40vh;

  --font-size: clamp(
    var(--min-font-size),
    var(--preferred-font-size),
    var(--max-font-size)
  );
}
</style>
