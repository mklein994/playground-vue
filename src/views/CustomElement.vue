<template>
  <custom-element></custom-element>
  <Highlightjs language="css" :code="styles" style="white-space: pre-wrap" />
</template>

<script lang="ts">
import hljsVuePlugin from "@highlightjs/vue-plugin";
import hljs from "highlight.js/lib/core";
import css from "highlight.js/lib/languages/css";
import { defineComponent, defineCustomElement } from "vue";

import CustomElement from "../components/CustomElement.ce.vue";

hljs.registerLanguage("css", css);

const Highlightjs = hljsVuePlugin.component;

const styles = CustomElement.styles[0];

customElements.define("custom-element", defineCustomElement(CustomElement));

export default defineComponent({
  name: "CustomElement",
  components: { Highlightjs },
  setup: () => {
    return {
      styles,
    };
  },
});
</script>

<style>
@import "highlight.js/styles/github-dark-dimmed.css";
</style>
