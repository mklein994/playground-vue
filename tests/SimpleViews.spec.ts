import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";

import CustomElement from "@/components/CustomElement.ce.vue";
import HomeView from "@/HomeView.vue";
import NotFound from "@/NotFound.vue";
import FooExperiment from "@/views/FooExperiment.vue";

describe.each([
  ["FooExperiment", FooExperiment, "Foo"],
  ["NotFound", NotFound, "404 Not Found"],
  ["HomeView", HomeView, "Welcome"],
  [
    "CustomElement",
    CustomElement,
    Array.from({ length: 10 }, (_, i) => `Item ${i + 1}`).join(""),
  ],
])("%s", (_name, component, text) => {
  it("renders correctly", () => {
    expect(component).toBeTruthy();
    const wrapper = mount(component);
    expect(wrapper.text()).toContain(text);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
