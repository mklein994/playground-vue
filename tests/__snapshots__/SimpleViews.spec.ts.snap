// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CustomElement > renders correctly 1`] = `
"<div class=\\"custom-element\\">
  <div class=\\"item\\">Item 1</div>
  <div class=\\"item\\">Item 2</div>
  <div class=\\"item\\">Item 3</div>
  <div class=\\"item\\">Item 4</div>
  <div class=\\"item\\">Item 5</div>
  <div class=\\"item\\">Item 6</div>
  <div class=\\"item\\">Item 7</div>
  <div class=\\"item\\">Item 8</div>
  <div class=\\"item\\">Item 9</div>
  <div class=\\"item\\">Item 10</div>
</div>"
`;

exports[`FooExperiment > renders correctly 1`] = `
"<h1>Foo</h1>
<p> Adipisicing nisi et explicabo voluptatum quisquam Quia expedita quae explicabo eveniet dolorum? Eligendi repellendus perferendis provident voluptatum similique Distinctio ipsam dignissimos vero sint aperiam rerum Veniam expedita quos fuga velit. </p>"
`;

exports[`HomeView > renders correctly 1`] = `
"<div data-v-c1f0f723=\\"\\" class=\\"welcome-container\\">
  <h1 data-v-c1f0f723=\\"\\" class=\\"welcome\\">Welcome</h1>
</div>"
`;

exports[`NotFound > renders correctly 1`] = `"<h1>404 Not Found</h1>"`;
